[Docs] Update code samples in docs (#527)
transparency-dev/trillian-tessera: README.md

    driver, err := mysql.New(ctx, db)
    driver, err := posix.New(ctx, dir, doCreate)

	appender, shutdown, reader, err := tessera.NewAppender(ctx, driver, tessera.NewAppendOptions().

> 		WithCheckpointSigner(s, a...))

Reviewer: Can probably drop the `a...` for the simple path?

The optional arguments that can be passed in allow Tessera behaviour to be tuned.

> Take a look at the functions in the `trillian-tessera` root package named `With*`, e.g. [`WithBatching`](https://pkg.go.dev/github.com/transparency-dev/trillian-tessera#WithBatching) to see the available options are how they should be used.

Reviewer: ```suggestion
Take a look at the methods named `With*` on the `AppendOptions` struct in the root package, e.g. [`WithBatching`](https://pkg.go.dev/github.com/transparency-dev/trillian-tessera#WithBatching) to see the available options are how they should be used.
```

The final part of configuring this storage object is to set up the addition features that you want to use.
These optional libraries can be used to provide common log behaviours.
See [Features](#features) after reading the rest of this section for more details.

#### Writing to the Log

Now you should have a storage object configured for your environment, and the correct features set up.
Now the fun part - writing to the log!

```go
func main() {

> 	appender, shutdown, reader, err := tessera.NewAppender(...)
>     if err != nil {
>       // exit
>     }
>     idx, err := appender.Add(ctx, tessera.NewEntry(data))()

Reviewer: ```suggestion
	appender, shutdown, reader, err := tessera.NewAppender(...)
	if err != nil {
		// exit
	}
	idx, err := appender.Add(ctx, tessera.NewEntry(data))()
```

Synchronous Integration is provided by [`tessera.IntegrationAwaiter`](https://pkg.go.dev/github.com/transparency-dev/trillian-tessera#IntegrationAwaiter).
This allows personality calls to `Add` to block until the new leaf is integrated into the tree.

## Lifecyles

### Appender

This is the most common lifecycle mode. Appender allows the personality to add leaves, which will be sequenced
contiguously with any prefix that the log has already committed to.

> This can be configured via [`tessera.NewAppender`](https://pkg.go.dev/github.com/transparency-dev/trillian-tessera@main#NewAppender).

Reviewer: ```suggestion
This mode can be configured via [`tessera.NewAppender`](https://pkg.go.dev/github.com/transparency-dev/trillian-tessera@main#NewAppender).
```

### Migration Target

This mode is used when either:

>  - the log is being operated as a mirror of another log

Reviewer: Currently it's only the 2nd of these - no checkpoint is ever written by the migration lib/tooling
